lib/eco/debug.py

```diff
@@ -0,0 +1,148 @@
+from time import sleep
+import telnetlib
+
+from PyQt4 import QtCore
+from PyQt4.QtCore import QObject
+from fcntl import fcntl, F_GETFL, F_SETFL
+from os import O_NONBLOCK, read
+from socket import error as socket_error
+
+class Debugger(QObject):
+    def __init__(self, window):
+        QObject.__init__(self)
+        self.window = window
+        # used from outside debugger
```

Comment:
Maybe change these comments to something like "incoming signals" / "outgoing signals".